src/Microsoft.AspNet.Razor/CodeGenerators/CodeWriter.cs

```diff
@@ -18,6 +19,8 @@ public class CodeWriter : IDisposable
         private int _currentLineIndex;
         private int _currentLineCharacterIndex;
 
+        public StringBuilder Builder => _writer.GetStringBuilder();
```

Comment:
1. There's a bunch of places we do `GenerateCode().Length`. It would be a lot nicer to simply do `Buffer.Length` instead. I'll send that as a follow up.
2. Don't see why it's better to move it a derived type.